src/modules/ekf2/EKF/python/ekf_derivation/derivation.py

```diff
@@ -729,6 +729,22 @@ def compute_gravity_z_innov_var_and_h(
 
     return (innov_var, H.T)
 
+def range_validation_filter() -> sf.matrix:
+
+    z = sf.Symbol("z")
+    dist_bottom = sf.Symbol("dist_bottom")
+    terrain = z - dist_bottom
+    state = sf.V2.symbolic("state")
```

Comment:
You can define the states in a `Values` structure like here: https://github.com/PX4/PX4-Autopilot/blob/f9140fcd503b9d01c9588ec8dda6d50359a08177/src/modules/ekf2/EKF/python/ekf_derivation/derivation.py#L63-L73

You can also pass a `Values` structure as an argument of `jacobian`